internal/cmd/common/flags.go

```diff
@@ -263,76 +264,88 @@ func HandleAttributeFlags(c *base.Command, suffix, fullField string, sepFields [
 		// First, perform any needed parsing if we are given the type
 		switch field.Name {
 		case "num-" + suffix:
-			// JSON treats all numbers equally, however, we will try to be a
-			// little better so that we don't include decimals if we don't need
-			// to (and don't have to worry about precision if not necessary)
-			if strings.Contains(field.Value, ".") {
-				val, err = strconv.ParseFloat(field.Value, 64)
+			switch {
+			case field.Value == nil:
+			case strings.Contains(field.Value.GetValue(), "."):
+				// JSON treats all numbers equally, however, we will try to be a
+				// little better so that we don't include decimals if we don't need
+				// to (and don't have to worry about precision if not necessary)
+				val, err = strconv.ParseFloat(field.Value.GetValue(), 64)
 				if err != nil {
 					return fmt.Errorf("error parsing value %q as a float: %w", field.Value, err)
 				}
-			} else {
-				val, err = strconv.ParseInt(field.Value, 10, 64)
+			default:
+				val, err = strconv.ParseInt(field.Value.GetValue(), 10, 64)
 				if err != nil {
 					return fmt.Errorf("error parsing value %q as an integer: %w", field.Value, err)
 				}
 			}
 
 		case "string-" + suffix:
-			val = strings.Trim(field.Value, `"`)
+			switch {
+			case field.Value == nil:
+			default:
+				val = strings.Trim(field.Value.GetValue(), `"`)
+			}
 
 		case "bool-" + suffix:
-			switch field.Value {
-			case "true":
-				val = true
-			case "false":
-				val = false
+			switch {
+			case field.Value == nil:
```

Comment:
Thinking about this, and looking at the new tests, is this what we want? This seems to turn a `boolean` into a ternary or a "[NullBool](https://pkg.go.dev/database/sql#NullBool)". I would think for this, and probably all of the type-prefixed ones, we should either error or force the value to be the specified type.

Reply:
That's fair. I can change it back. For the same reason string-attr should not be key only and num-attr, yes?

Reply:
> For the same reason string-attr should not be key only and num-attr, yes?

Yea, I think so.

Reply:
Done!